design/one-pager-class-selection.md

```diff
@@ -0,0 +1,371 @@
+hypothetical GitLab [stack]. It declares a need for an experimental grade
+PostgreSQL instance in region us-east-1 that is suitable for use with the GitLab
+stack. Any compatible resource class (whether it be an `RDSInstanceClass` or a
+`CloudSQLInstanceClass`) matching the resource claim's labels would be eligible
+to satisfy it. If more than one such class existed one would be selected at
+random. A resource claim that omitted `.spec.classSelector`, `.spec.classRef`,
+and `.spec.resourceRef` would be deemed to be completely unopinionated and would
+match a compatible resource class at random.
+
+This pattern is likely familiar to Kubernetes and to Crossplane users - it's
+used to select which nodes a pod may run on, which pods are endpoints of a
+service, and which Kubernetes cluster a Crossplane workload may run on.
+
+### Matching Scope
+
+Resource classes would be selected only from within the same namespace as the
+resource claim by default; this is the simplest, safest, and least surprising
+default for small scale Crossplane deployments. The infrastructure namespace
+pattern would be enabled by creating an association between application and
+infrastructure namespaces using namespace level annotations. For example:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: acme-team
+  annotations:
+    resource-class.crossplane.io/select-namespaces: "acme-team, experimental-infra"
+```
+
+The above `acme-team` application namespace declares that any resource claims
+created within it should be matched against resource classes from namespaces
+`acme-team` or `experimental-infra`. Note that when the `select-namespaces`
+annotation exists it overrides the default behaviour of selecting only within
+the same namespace as the resource claim, and thus the namespace must declare
+itself to be in scope for class selection if it so desires. A special wildcard
+namespace `*` would allow matching across any namespace, for example:
+`resource-class.crossplane.io/select-namespaces: "*"`.
+
+### Exact Matching and Static Provisioning
+
+Resource claims would retain their existing `.spec.classRef` field under this
+proposal. In fact the `.spec.classSelector` would be used to set the `classRef`,
+just like Crossplane's [workload scheduling]. Resource claim authors who desired
+dynamic provisioning but had a specific resource class in mind could bypass the
+label selection process by explicitly referencing a resource class, for example:
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  namespace: acme-team
+  name: gitlab-database
+spec:
+  classRef:
+    apiVersion: database.gcp.crossplane.io/v1alpha2
+    kind: CloudSQLInstance
+    namespace: us-east-1
+    name: gitlab-experimental
+  writeConnectionSecretToRef:
+    name: app-postgresql-connection
+```
+
+Static provisioning is unaffected by this proposal; setting the
+`.spec.resourceRef` field of a resource claim would continue to bypass resource
```

Comment:
Is `.spec.resourceRef` an alias `.spec.classRef` in the example? If so, a clarification could help the reader.

Reply:
Added some context at the first mention of these fields.